src/CommunityToolkit/Xamarin.CommunityToolkit/Views/CameraView/iOS/FormsCameraView.ios.cs

```diff
@@ -305,7 +305,13 @@ public void FinishedRecording(AVCaptureFileOutput captureOutput, NSUrl outputFil
 
 		public void SwitchFlash(CameraFlashMode newFlashMode)
 		{
-			if (isAvailable && device != null && newFlashMode != flashMode)
+			if (!isAvailable || device == null || newFlashMode == flashMode)
+			{
+				return;
+			}
+
+			if ((flashMode == CameraFlashMode.Torch && device.TorchAvailable)
+				|| (flashMode != CameraFlashMode.Torch && device.FlashAvailable))
```

Comment:
```suggestion
			var isSupported = newFlashMode == CameraFlashMode.Torch
				? device.TorchAvailable
				: device.FlashAvailable;

			if (isSupported && isAvailable && device != null && newFlashMode != flashMode)
```

Reply:
I guess we should compare **newFlashMode**
Please consider my proposed solution